src/main/java/com/autotune/analyzer/recommendations/engine/CostRecommendationEngine.java

```diff
@@ -780,75 +780,160 @@ private boolean populateRecommendation(String recommendationTerm,
 
         // Check for thresholds
         if (isRecommendedCPURequestAvailable) {
-            if (isCurrentCPURequestAvailable && currentCpuRequestValue > 0.0) {
+            if (isCurrentCPURequestAvailable && currentCpuRequestValue > 0.0 && null != generatedCpuRequest) {
```

Comment:
there are 3 conditions. Is there a way to simplify them? PS: This is just a thought and may not be a valid comment.